pkg/kev/dev.go

```diff
@@ -33,17 +33,28 @@ import (
 )
 
 // NewDevRunner creates a render runner instance
-func NewDevRunner(workingDir string, handler ChangeEventHandler, opts ...Options) *DevRunner {
-	runner := &DevRunner{chgEventHandler: handler, Project: &Project{workingDir: workingDir}}
+func NewDevRunner(workingDir string, opts ...Options) *DevRunner {
+	runner := &DevRunner{
+		Project: &Project{
+			WorkingDir: workingDir,
+			eventHandler: func(e RunnerEvent, r Runner) error {
+				return nil
+			},
+		},
+	}
 	runner.Init(opts...)
-	if runner.config.skaffold && len(runner.config.k8sNamespace) == 0 {
-		runner.config.k8sNamespace = DefaultSkaffoldNamespace
+	if runner.config.Skaffold && len(runner.config.K8sNamespace) == 0 {
+		runner.config.K8sNamespace = DefaultSkaffoldNamespace
 	}
 	return runner
 }
 
 // Run runs the dev command business logic
 func (r *DevRunner) Run() error {
+	if err := r.eventHandler(DevLoopStarting, r); err != nil {
+		return errors.Errorf("%s\nwhen handling fired event: %s", err.Error(), DevLoopStarting)
```

Comment:
would it be better if we said which event that was in the message? So `when handling fired event` would become `dev loop couldn't start` or something along these lines?